src/general-components/UniqueCheck/UniqueCheck.tsx

```diff
@@ -140,12 +130,64 @@ export class UniqueCheck extends Component<ReplaceProps<"input", FormControlProp
                     )}
 
                     {
-                        !this.props.suppressErrors && this.renderErrors()
+                        !suppressErrors && this.renderErrors()
                     }
 
                 </div>
             </>
         );
     }
 
+
+    changed = async (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        if (this.timeout) {
+            clearTimeout(this.timeout);
+        }
+
+        let value = e.currentTarget.value;
+        if (value.length > 0) {
+            this.timeout = setTimeout(async () => {
+                this.setState({
+                    isLoading: true
+                });
+
+                try {
+                    // wrap the errorCallback in a Promise, so it can get awaited
+                    let call = await new Promise<CallInterface<DefaultResponse<AvailabilityCheckResource>> | null>((resolve, reject) => {
+                        if (this.props.callback === undefined) {
+                            reject(new Error("callback is undefined"));
+                        } else {
+                            this.props.callback(value, {errorCallback: reject}).then(resolve).catch(reject);
+                        }
+                    });
+
+
+                    if (call?.success) {
+                        this.setState({
+                            success: call.callData.data.available,
+                            reason: call.callData.data.reason,
+                            isLoading: false,
+                            error: false
+                        });
+                        return;
+                    }
+                } catch (reason) {
+                    console.dir(reason);
```

Comment:
remove this

Reply:
added check if react is in debug mode and only prints if it is